### PR TITLE
Generate token at signup

### PR DESCRIPTION
**Pull request by jamesmcleish**
This pull request adds a token to the user at signup. This means that the user will have a token when they go to bedroom.

### CODE REVIEW

File: backend/src/routes/api.js
Comment (jamesmcleish) [inline backend/src/routes/api.js:1]: token checker added to stats posting

File: backend/src/util/tokenChecker.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const tokenChecker = (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (authHeader) {
+    const token = authHeader.split(' ')[1];
+    JWT.verify(token, 'SECRET', (err, decoded) => {
+      if (err) {
+        return res.status(401).json({ error: 'Token is not valid' });
+      } else {
+        req.decoded = decoded;
+        next();
```
Comment (jamesmcleish) [inline backend/src/util/tokenChecker.js:12]: next(); to move onto next function in the chain e.g. gameRouter

File: backend/src/util/tokenChecker.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const JWT = require("jsonwebtoken");
+
+const tokenChecker = (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (authHeader) {
+    const token = authHeader.split(' ')[1];
+    JWT.verify(token, 'SECRET', (err, decoded) => {
```
Comment (jamesmcleish) [inline backend/src/util/tokenChecker.js:7]: JWT function that takes the user token and our sercret key 'SECRET' and sees if they match.

File: frontend/src/components/EndGame/EndGame.component.jsx [hunk 3]
```diff
@@ -37,10 +37,12 @@ const EndGame = () => {
   };
 
   const storeResult = async () => {
+    const token = window.localStorage.getItem('token')
     const response = await fetch("http://localhost:8000/game/update", {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
+        Authorization: `Bearer ${token}`,
```
Comment (jamesmcleish) [inline frontend/src/components/EndGame/EndGame.component.jsx:45]: add token to post request to fulfill tokenChecker requirements

File: frontend/src/components/EndGame/EndGame.component.jsx
Comment (edwardfgt) [inline frontend/src/components/EndGame/EndGame.component.jsx:1]: Very cool, I created an account and then deleted my tokens and tried to play through a game and it wouldn't allow me to finish the game so the verification seems to be working well.

**Review by edwardfgt — APPROVED** [review]
Good job, I tested the code and seems to work well. I think we can add this to other parts of the site easily too.

**Review by SimpleLuke — APPROVED** [review]
It looks good to me. I updated some backend test codes.

**Pull request merged**